src/components/MealPage.js

```diff
@@ -0,0 +1,24 @@
+import { Typography, Button } from "@mui/material";
+import { useParams } from "react-router-dom";
+import IngridientsList from "./IngridientsList";
+import { Link } from "react-router-dom";
+
+export default function MealPage({ mealsData }) {
+  const params = useParams();
+
+  const currentMeal = mealsData.find(
+    (meal) => meal.id === Number.parseInt(params.id)
```

Comment:
1. `Number.parseInt` can return `NaN`.  Consider handling it explicitly.
2. Don't run `parseInt` on every interaction. Precalculate it before calling `find`.